controller/main.go

```diff
@@ -74,16 +75,26 @@ func (c *controller) SetBalancer(l log.Logger, name string, svcRo *v1.Service, _
 	// a reason.
 	svc := svcRo.DeepCopy()
 	syncStateRes := controllers.SyncStateSuccess
+
+	var prevIPs []net.IP
 	wasAllocated := c.isServiceAllocated(name)
+	if wasAllocated {
+		prevIPs = c.allocatedIPs(name)
+	}
+
 	if c.convergeBalancer(l, name, svc) != nil {
 		syncStateRes = controllers.SyncStateErrorNoRetry
 	}
 
-	if wasAllocated && !c.isServiceAllocated(name) { // convergeBalancer may deallocate our service and this means it did it.
-		// if the service was deallocated, it may have have left room
-		// for another one, so we reprocess
-		level.Info(l).Log("event", "serviceUpdated", "msg", "removed loadbalancer from service, services will be reprocessed")
-		syncStateRes = controllers.SyncStateReprocessAll
+	if wasAllocated && !c.isServiceAllocated(name) {
```

Comment:
not related to this pr, but the deepequal below should probably go before this.

Reply:
indeed

Reply:
Let's do it then :-) A separate commit in this PR is fine, even if it's out of scope.

What we normally do is, if we find some cleanup out of scope, it is fine to do it contextually as long as it belongs to a separate commit. This is possible because we don't squash so each entry will be separated.

Reply:
ok

Reply:
done